### PR TITLE
Labels query performance improvement

### PR DESCRIPTION
**Pull request by mcovarr**
One representative pair of timings on a stone cold prod clone (restarted between queries) for a single submission ID label with two ORed collection labels:

develop: 31.20 sec
this branch: 0.08 sec

No index changes required. MySQL will use a k/v index on `CUSTOM_LABEL_ENTRY` if it's available but its presence doesn't seem to actually make much difference in execution time.

Exclude labels have been folded into the new system, I still need to measure the performance impact of those changes.

### CODE REVIEW

File: database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala [hunk 1]
```diff
@@ -123,20 +111,79 @@ trait WorkflowMetadataSummaryEntryComponent {
...
+      endTimestampOption,
+      includeSubworkflows
+    )
+
+    // First do the summary table filtering, then some performance optimizations below for labels.
+    val baseQuery = workflowMetadataSummaryEntries.filter(filter)
+
+    // Label ANDs and ORs can be expressed as `join`s which at least on MySQL appear to execute more efficiently than
+    // equivalent `exists` formulations.
+    type LabelsQuery = Query[CustomLabelEntries, CustomLabelEntry, Seq]
```
Comment (cjllanwarne) [inline database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala]: Why `Seq` here and `Set` below?

**Comment by danbills** [comment]
:+1: but curious what excludes perf looks like.

File: database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala (hunk 1 above)
Comment (mcovarr) [reply database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala]: The label kv inputs are `Set`s and the `Query`s collections parallel those. The `Seq`s come from the Slick API.

File: database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala [hunk 2]
```diff
@@ -94,22 +90,14 @@ trait WorkflowMetadataSummaryEntryComponent {
       val workflowStatusFilter = workflowStatuses.
         map(workflowStatus => workflowMetadataSummaryEntry.workflowStatus.fold(ifEmpty = exclude)(_ === workflowStatus)).
         reduceLeftOption(_ || _)
-      val labelsAndFilter = existsWorkflowLabels(workflowMetadataSummaryEntry, labelAndKeyValues, _ && _)
-      val labelsOrFilter = existsWorkflowLabels(workflowMetadataSummaryEntry, labelOrKeyValues, _ || _)
-      val excludeLabelsAndFilter = existsWorkflowLabels(workflowMetadataSummaryEntry, excludeLabelAndValues, _ && _).map(v => !v)
-      val excludeLabelsOrFilter = existsWorkflowLabels(workflowMetadataSummaryEntry, excludeLabelOrValues, _ || _).map(v => !v)
       val notASubworkflowFilter: Option[Rep[Boolean]] =
-        if (includeSubworkflows) None else Some(!metadataEntryExistsForWorkflowExecutionUuid(workflowMetadataSummaryEntry.workflowExecutionUuid, parentIdWorkflowMetadataKey))
+        if (includeSubworkflows) None else Option(!metadataEntryExistsForWorkflowExecutionUuid(workflowMetadataSummaryEntry.workflowExecutionUuid, parentIdWorkflowMetadataKey))
```
[the remark this answers is not in the document]
Comment (mcovarr) [reply database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala]: For the separate #4561 issue, yes

**Pull request merged**